gke-deploy/core/gcs/gcs.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2019 Google, Inc. All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// Package gcs contains logic related to Google Cloud Storage.
+package gcs
+
+import (
+	"context"
+	"errors"
+	"log"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/services"
+)
+
+const TimeoutErr = "GCS timeout"
+
+var (
+	defaultTimeout = 1 * time.Hour
+	errGCSTimeout  = errors.New(TimeoutErr)
+)
+
+type GCS struct {
+	Timeout    time.Duration
```

Comment:
Can these fields be private or do you plan to use them outside of this package?

Reply:
My idea is those variable should be changeable after GCS instance was created. E.g, Retries could be modify dynamically if the failure rate is high in certain time. But it is debatable since we don't have such requirements in this project. Do you think it is better to hide them?

Reply:
If we don't expect to need to change them I would avoid "future proofing". You should always minimize visibility where possible. In the future if that functionality is needed, we can expose it then.

Reply:
Done